### DISCUSSION TITLE
How can I redirect `print` to output

### DISCUSSION THREAD

**Discussion by LukeLIN-web** [facebookresearch/hydra#2568] [Q&A] [answered]
I am using `
@hydra.main(config_path='conf', config_name='config', version_base='1.1')`
However, the ouput file is empty, how can I store `print` to output file ?

**Comment by omry** [comment]
The output log file is for python logging usage.
You might find some unofficial Python hack to redirect stdout to a logger (like [this](https://stackoverflow.com/questions/19425736/how-to-redirect-stdout-and-stderr-to-logger-in-python) one) but in general the recommendation is that you use the logging API directly.

See [this](https://hydra.cc/docs/tutorials/basic/running_your_app/logging/) page.

**Reply by LukeLIN-web** [reply]
> The output log file is for python logging usage. You might find some unofficial Python hack to redirect stdout to a logger (like [this](https://stackoverflow.com/questions/19425736/how-to-redirect-
![Uploading Screen Shot 2023-02-03 at 10.49.01 AM.png…]()
stdout-and-stderr-to-logger-in-python) one) but in general the recommendation is that you use the logging API directly.
> 
> See [this](https://hydra.cc/docs/tutorials/basic/running_your_app/logging/) page.

Yes, I use `logging`, but the log still can not be stored in the output file.

**Reply by Jasha10** [reply]
@LukeLIN-web how are you using logging?
Have you tried the Hydra logging tutorial?
https://hydra.cc/docs/1.3/tutorials/basic/running_your_app/logging/

**Reply by LukeLIN-web** [reply]
> @LukeLIN-web how are you using logging? Have you tried the Hydra logging tutorial? https://hydra.cc/docs/1.3/tutorials/basic/running_your_app/logging/

I am sorry, I know why. 
the log still can not be stored in the output file  because this line is in subprocess.
Did you have any idea how to output it?
```python
def run_sample():
    log.info("Info level message")

@hydra.main(config_path='conf', config_name='config', version_base='1.1')
def main(conf):
    for worker_id in range(num_sample_worker):
        p = mp.Process(target=run_sample, args=(
            worker_id, params, dataset, quiver_sampler, microbatchs_qs))
        p.start()
        workers.append(p)
```

**Comment by omry** [comment, marked as answer]
You probably need to initialize the logging in the subprocess.
There is a function for it in Hydra somewhere that takes the Hydra config object and does that.
